src-gui/src/renderer/components/modal/seed-selection/SeedSelectionDialog.tsx

```diff
@@ -83,6 +105,10 @@ export default function SeedSelectionDialog() {
     }
   };
 
+  const isBlockheightValid = parseBlockHeightInput(blockheightInput) !== false;
+  const isBlockheightInputFailed =
```

Comment:
Related to the comment above about the `string | undefined`. When you hover over the name of this variable you can see that it is `boolean | "" | undefined`. I believe this is because this expression (`blockheightInput && isBlockheightValid === false`) might return early or something along those lines. JavaScript can be a bit finnicky.

When you change `blockheightInput` to be of type `string` with a default value of `""`, you can change this to `blockHeightInput.length > 0 && isBlockheightValid === false`. The type of it will then be simple `boolean` (which is what we want).